### PR TITLE
Load balancer

### PR DESCRIPTION
**Pull request by shaunclarke2333**

### CODE REVIEW

File: level2/vpc-data-source.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#This remote data source is for main vpc and main-public0 id
+data "terraform_remote_state" "main-vpc" {
```
Comment (vladimirmukhin) [inline level2/vpc-data-source.tf]: It's better to call this datasource level1

File: level2/securitygroup.tf [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  egress {
+    from_port   = 0
+    to_port     = 65535
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = "main-allows-ssh"
```
Comment (vladimirmukhin) [inline level2/securitygroup.tf:24]: This sg allows both 22 and 80.
Which means `main-allow-ssh` is not an accurate name

File: level2/securitygroup.tf [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    protocol    = "tcp"
+    cidr_blocks = ["70.95.138.224/32"]
+
+  }
+  ingress {
+    description = "http to VPC"
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
```
Comment (vladimirmukhin) [inline level2/securitygroup.tf]: You don't need to open 80 everywhere in this security group. Instead, you can allow connections on port 80 from the security group of your load balancer

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

File: level2/vpc-data-source.tf (hunk 1 above)
Comment (shaunclarke2333) [reply level2/vpc-data-source.tf]: added "level1 to the data source name, making it level1-main-vpc

File: level2/securitygroup.tf (hunk 2 above)
Comment (shaunclarke2333) [reply level2/securitygroup.tf:24]: removed port 80 inbound rule from this security group.

File: level2/securitygroup.tf (hunk 3 above)
Comment (shaunclarke2333) [reply level2/securitygroup.tf]: removed the port 80 inbound rule and added the load balancer security group to the ec2 instance

**Review by vladimirmukhin — APPROVED** [review]

**Pull request merged**